actions/maven/publish/action.yml

```
@@ -0,0 +1,117 @@
+  provenance-download-sha256:
+    description: "The sha256 of the package provenance artifact."
+    required: false
+    type: string
+  target-download-sha256:
+    description: "The sha256 of the target directory."
+    required: true
+    type: string
+  maven-username:
+    description: "Maven username"
+    required: false
+  maven-password:
+    description: "Maven password"
+    required: false
+  gpg-key-pass:
+    description: "gpg-key-pass"
+    required: false
+  gpg-private-key:
+    description: "gpg-key-pass"
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the project repository
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-project-checkout@v1.8.0-rc.1
+    - name: Set up Java for publishing to Maven Central Repository
+      uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3
+      env:
+        MAVEN_USERNAME: ${{ inputs.maven-username }}
+        MAVEN_PASSWORD: ${{ inputs.maven-password }}
+        GPG_KEY_PASS: ${{ inputs.gpg-key-pass }}
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+        gpg-private-key: ${{ inputs.gpg-private-key }}
+        gpg-passphrase: GPG_KEY_PASS
+
+    - name: Download the slsa attestation
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@v1.8.0-rc.1
+      with:
+        name: "${{ inputs.provenance-download-name }}"
+        path: slsa-attestations
+        sha256: "${{ inputs.provenance-download-sha256 }}"
+
+    - name: Download the target dir
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@v1.8.0-rc.1
+      with:
+        name: target
+        path: ./
+        sha256: "${{ inputs.target-download-sha256 }}"
+
+    - name: Publish to the Maven Central Repository
+      shell: bash
+      env:
+        MAVEN_USERNAME: "${{ inputs.maven-username }}"
+        MAVEN_PASSWORD: "${{ inputs.maven-password }}"
+        GPG_KEY_PASS: "${{ inputs.gpg-key-pass }}"
+        SLSA_DIR: "${{ inputs.provenance-download-name }}"
+        PROVENANCE_FILES: "${{ inputs.provenance-download-name }}"
+      run: |
+        # Build and run custom plugin
+        cd plugin && mvn clean install && cd ..
+        # Re-indexing the secondary jar files for deploy
+        mvn javadoc:jar source:jar
+        # Retrieve project version
+        version=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.version -q -DforceStdout)
+        ARTIFACTID=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.artifactId -q -DforceStdout)
+        # Reset the environment variables add in the base provenance
+        FILES="slsa-attestations/${PROVENANCE_FILES}/${ARTIFACTID}-${version}.jar.intoto.build.slsa"
+        TYPES=slsa
+        CLASSIFIERS=jar.intoto.build
```

Use lowercase variable names for script-local variables (to distinguish from those defined in `env` etc.)

```suggestion
        artifact_id=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.artifactId -q -DforceStdout)
        # Reset the environment variables add in the base provenance
        files="slsa-attestations/${PROVENANCE_FILES}/${artifact_id}-${version}.jar.intoto.build.slsa"
        types=slsa
        classifiers=jar.intoto.build
```